src/main/java/com/coveros/selenified/services/Check.java

```diff
@@ -270,6 +321,95 @@ boolean checkContains(JsonElement expectedJson) {
+     * containing each of the pair values provided. The jsonKeys should be passed in
+     * as crumbs of the keys leading to the field with
+     * the expected value. This result will be written out to the output file.
+     *
+     * @param jsonCrumbs    - the crumbs of json object keys leading to the field with the expected value
+     * @param expectedPairs a hashmap with string key value pairs expected in the json
+     *                      response
+     */
+    boolean checkContains(List<String> jsonCrumbs, Map<String, Object> expectedPairs) {
+        JsonObject actualValue = this.response.getObjectData();
+        for (String jsonCrumb : jsonCrumbs) {
+            if (actualValue == null) {
+                break;
+            }
+            if (actualValue.get(jsonCrumb) instanceof JsonObject) {
+                actualValue = actualValue.get(jsonCrumb).getAsJsonObject();
+            } else {
+                actualValue = null;
+                break;
+            }
+        }
+        boolean pass = doesJsonObjectContainPair(expectedPairs, actualValue);
+        if (pass) {
+            this.reporter.pass("", EXPECTED_TO_FIND + STARTI + Reporter.formatHTML(String.join(" \uD83E\uDC1A ", jsonCrumbs)) + ENDI + " containing: " + DIV_I + Reporter.formatKeyPair(expectedPairs) + END_IDIV, FOUND + DIV_I + Reporter.formatHTML(GSON.toJson(actualValue)) + END_IDIV);
+        } else {
+            this.reporter.fail("", EXPECTED_TO_FIND + STARTI + Reporter.formatHTML(String.join(" \uD83E\uDC1A ", jsonCrumbs)) + ENDI + " containing: " + DIV_I + Reporter.formatKeyPair(expectedPairs) + END_IDIV, FOUND + DIV_I + Reporter.formatHTML(GSON.toJson(actualValue)) + END_IDIV);
+        }
+        return pass;
+    }
+
+    /**
+     * Checks the actual response json payload contains a key containing a JsonElement.
+     * The jsonKeys should be passed in as crumbs of the keys leading to the field with
+     * the expected value. This result will be written out to the output file.
+     *
+     * @param jsonCrumbs   - the crumbs of json object keys leading to the field with the expected value
+     * @param expectedJson - the expected response json array
+     */
+    @SuppressWarnings("squid:S1201")
+    abstract void contains(List<String> jsonCrumbs, JsonElement expectedJson);
+
+    /**
+     * Checks the actual response json payload contains a key containing a JsonElement.
+     * The jsonKeys should be passed in as crumbs of the keys leading to the field with
+     * the expected value. This result will be written out to the output file.
+     *
+     * @param jsonCrumbs   - the crumbs of json object keys leading to the field with the expected value
+     * @param expectedJson - the expected response json array
+     */
+    boolean checkContains(List<String> jsonCrumbs, JsonElement expectedJson) {
+        JsonElement actualValue = this.response.getObjectData();
+        for (String jsonCrumb : jsonCrumbs) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS135)